get_deps.py

```diff
@@ -30,11 +37,23 @@ def main():
 		if dll_name.startswith('???'):
 			print('Unknown DLL?')
 			continue
-		if dll_path.lower().startswith('/c/windows') or dll_path.lower().startswith('c:/windows'):
+		if has_cygpath:

```

Comment:
I don't understand why we continue on if cygpath isn't available... if we don't have cygpath shouldn't we just fail? Because otherwise (according to the comment) we would be incorrectly feeding UNIX style paths into Python.

...but more generally I haven't looked into this issue on my own machine recently, but this originally used to work for me and on the CI system. I'm still guessing that *somehow* native Python is being invoked (despite `which` showing msys2 version) to call this script. Notice how in the appveyor.bat script we explicitly update `PATH` to include msys2 binaries first.

Reply:
This is bikeshedding by now. Let's stick to any of the proposed solutions which work (this, or the original version which just resulted in a backtrace).

I brought up the case of cygpath not existing as older versions of MSYS2 did not ship it. There are probably non-MSYS2/CYGWIN environments which could be used to compile QEMU.
I felt getting a best-effort approach or seeing a meaningful error message would be better than seeing a backtrace in those cases.

I'll test which python is being ran tomorrow *again*. Also keep in mind that MSYS has different versions of python for different purposes. In my opinion, either version of python should work, so using cygpath sounds good to me (with cygwin being a popular / recommended environment).

*(Note that I did not test this last revision, and I'd like to test it before merge, because I'm affected by this issue and want to be sure that this fixes it)*

Reply:
I don't know much about how the folder layout on Appveyor looks like, but here's my guess about why it works.
- In the mingw64 shell (NOT the msys2 shell), `os.path.abspath('/mingw64/test')` results in `C:/mingw64/test`, which is wrong. Strangely enough, in the msys2 shell the result is correct (`/mingw64/test`).
- ldd gives back paths like `/mingw64/bin/libffi-6.dll`. Based on the above, Python will interpret this path as `C:/mingw64/bin/libffi-6.dll`. For me, this is wrong in on two accounts. First, msys2 is installed in `C:/msys64`, and secondly, I'm building on disk D, so Python actually interprets the path as `D:/mingw64/bin/libffi-6.dll`.
- Based on `appveyor.bat`, Appveyor only installs python2 but not mingw-w64-x86_64-python2 (which I have installed too), so it uses the Python version that converts paths correctly and works fine.
Regarding the absence of cygpath, I don't really feel strongy one way or the other. I guess it depends on how many scenarios and build environments we try to support.

Reply:
I was confused about why CI builds were working before and after this patch, so I've done a little testing on my Windows 10 VM with the MINGW64 MSYS2 shell with the MSYS2 build of Python 2. Interestingly, it appears to handle paths in both the UNIX style format **and** the Windows native path format. But I'm guessing the version of Python that you have installed is a little pickier and only accepts the Windows style paths.

Some version info and testing:
```
$ python2
Python 2.7.13 (default, Feb 14 2017, 14:46:01)
[GCC 6.3.0] on msys
Type "help", "copyright", "credits" or "license" for more information.
>>> f = open(r'C:\Windows\notepad.exe', 'rb')
>>> len(f.read())
245760
>>> f2 = open('/c/Windows/notepad.exe', 'rb')
>>> len(f2.read())
245760
>>> import os.path
>>> os.path.abspath('/c/Windows/notepad.exe')
'/c/Windows/notepad.exe'
>>> os.path.abspath(r'C:\Windows\notepad.exe')
'C:\\Windows\\notepad.exe'
>>>
``` 

So this explains why it seems that CI is working before/after this PR. For this reason I'm okay with just using `cygpath` to convert paths to Windows format.

As for the missing `cygpath` issue, let's just print an error message and `exit(1)` if we fail to run `cygpath`. If the situation arises where we need to support an environment without `cygpath` in the future, we can solve it then.

Thanks!

Reply:
For the record, this is the output in the mingw64 shell after installing `mingw-w64-x86_64-python2`:
```
$ python2
Python 2.7.15 (default, May  4 2018, 07:39:01)  [GCC 7.3.0 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os.path
>>> os.path.abspath('/c/windows.notepad.exe')
'C:/c/windows.notepad.exe'
>>> os.path.abspath('c:\\windows\\notepad.exe')
'c:/windows/notepad.exe'
>>>
```